### DISCUSSION TITLE
Engenharia de Software ou Ciência da Computação?

### DISCUSSION THREAD

**Discussion by renanpvaz** [frontendbr/forum#713] [Discussão]
Estou tentando decidir qual curso deveria fazer. Portanto, gostaria de ouvir a opinião de quem faz/fez algum desses cursos.

**Comment by tdcosta100** [comment]
Olá, renanpvaz. Sou formado em Ciência da Computação, então vou falar um pouco do que eu decidi na época em que fui fazer minha graduação. As duas áreas têm bastante em comum, mas têm suas ênfases. Claro que tem um bastante de hardware em Ciência da Computação e bastante de software em Engenharia da Computação. Mas o foco de Engenharia da Computação é hardware, e o de Ciência da Computação é software.

Ao decidir por Ciência da Computação, entendi que gostaria de me dedicar mais ao software do que ao hardware, e não me arrependo nem um pouco. Amo muito o que faço. Você, criando esse tópico em uma discussão de Front-End, talvez tenha mais afinidade com Ciência da Computação. Mas nada impede que faça Engenharia da Computação e seja um excelente front-ender, por exemplo.

Uma vantagem a se considerar ao se formar em Engenharia da Computação é você se formar engenheiro, ou seja, terá CREA e tudo, e poderá eliminar várias matérias caso queira cursar outra graduação de engenharia.

Em questões de mercado de trabalho, creio haver espaço pra todo mundo, mas pra Ciência da Computação o mercado me parece um pouco mais dinâmico e receptivo. Porém são minhas impressões, a realidade pode diferir um pouco.

Procure dar uma olhada na grade curricular de cada um, e ler um pouco sobre a proposta também. Os sites das faculdades são uma noção boa sobre isso.

No caso do meu curso, a grade curricular foi essa (ingressei em 2004 #old):

http://www.ibilce.unesp.br/#!/graduacao/cursos/ciencia-da-computacao/estrutura-curricular/

Abraços e boa sorte na escolha da carreira!

**Comment by renanpvaz** [comment]
Olá @tdcosta100, agradeço a resposta, ajudou a entender melhor. Ainda assim, preciso dizer que na verdade referi-me ao curso de Engenharia de **Software**, e o fiz justamente por ser aparentemente similar ao outro curso.

**Comment by tdcosta100** [comment]
Putz, você tem razão hahaha. Falha minha. Bom, Ciência da Computação é um curso bem mais abrangente que Engenharia de Software, embora ambos sejam realmente bem parecidos. No meu curso de Ciência da Computação vi tópicos de Sistemas de Computação (basicamente funcionamento de sistemas operacionais e seus inúmeros mecanismos internos), Sistemas de Automação e Controle Digital (isso aqui é praticamente dois pés dentro de Engenharia da Computação), Computação Científica (pra quem gosta da área acadêmica é prato cheio) e Sistemas de Informação (que seria o equivalente ao seu curso de Engenharia de Software).

Engenharia de Software parece mais voltado ao mercado, te deixará mais pronto pra ele, talvez. Porém Ciência da Computação abrange mais teoria, fora toda a base matemática que cursos de matemática e engenharia têm (Cálculo Diferencial e Integral, Geometria Analítica e Vetores, Álgebra Linear, etc). Vai da sua escolha a questão do tempo de formação (Engenharia de Software é 3 anos, não? Ciência da Computação são pelo menos 4), e o aprofundamento que você quer ter nas teorias que originaram a computação.

**Reply by Alfredo-Fontinele** [reply]
A média da Engenharia de Software é 4 anos também. E é isso mesmo que você escreveu, o foco dela, assim como Análise e Desenvolvimento de Sistemas, é o mercado de trabalho, mas no caso de ADS é um curso tecnólogo. Já a Ciência da Computação e Engenharia de Software, ambas são bacharelado com regulamentação no CREA.

**Reply by tdcosta100** [reply]
Será que você não está se referindo a Engenharia da Computação? Porque fiz Ciência da Computação, mas não posso ter CREA por não ser engenharia, embora tenha várias matérias de engenharia no currículo.

**Comment by safirelauene** [comment]
Oi renanpvaz, legal você estar procurando uma formação superior na área de TI. Não fiz nenhum desses dois cursos, mas fiz um tecnólogo que também vale como curso superior. 
Um tecnólogo varia de 2 a 3 anos e procura fazer com que os estudos sejam abordados de uma forma mais prática, isso faz com que o aluno saia mais preparado para o mercado de trabalho. 
Fiz o curso de Tecnologia em Banco de Dados pela FATEC SJC e me formei final do ano passado,. Trabalho como frontend faz um ano e no segundo ano da faculdade já dá pra começar um estágio. Caso você esteja procurando algo desse tipo, sugiro um tecnólogo. :)

**Reply by devpro999** [reply]
me start in 2024 13 years old

**Comment by khaueviana** [comment]
Tecnólogo em Análise e Desenvolvimento :)

**Comment by weltonsanches** [comment]
@renanpvaz Eu fiz Tecnólogo em Análise e Desenvolvimento de Sistemas. Mas pelo que conheço dos cursos de Engenharia de Software e Ciência da Computação vai ser mais ou menos isso:

- **Ciência da Computação** você vai mais a fundo no Code, seus fundamentos e tudo o que precisa pra que ele rode (uma base matemática mais forte, Sistemas Operacionais, compiladores e tals).

- **Engenharia de Software** você vai focar mais no processo de desenvolvimento de software (ciclo de vida, metodologias e etc).

O melhor curso? Depende do seu objetivo profissional. Nada impede que mude lá na frente, já que ambos terão um pouco do outro. Ver as disciplinas e ementa destas disciplinas no site da faculdade é boa ideia.

**Comment by yanmagale** [comment]
Eu fiz _Ciência da Computação_ e é bem como o @WeltonSanches disse. 

Na primeira metade do curso, você terá muitas disciplinas voltadas a matemática e física. No restante do curso você já tem disciplinas mais voltadas a parte de compreensão da Computação, como Grafos, Compiladores, Máquina de Turing, Análise de Algoritmos e etc. A grande verdade é o que o curso te prepara mesmo para ser um pesquisador. 

Antes de entrar na faculdade eu tinha feito um curso técnico, onde fiquei 1 ano e meio  aprendendo sobre programação. Então quando iniciei a faculdade, eu já era um programador, então algumas coisas não foi tanta novidade assim. Como eu já tinha esta base, e quando eu realmente havia decido que queria programar mais e mais, entender mais a fundo como as coisas funcionavam, a minha escolha foi direta no curso de Ciências da Computação. 

O curso me ajudou em algumas coisas (e ajuda), mas conta muito mais as coisas que você faz fora dele: você estudar as novas tecnologias, praticar novas técnicas, testar conceitos vistos em sala de aula.

Espero ter ajudado :smile:

**Comment by wagber** [comment]
De forma bem resumida posso dizer que a pessoa qu decide cursar Engenharia de Software é a pessoa que gosta de criar, inovar, fazer parte de times onde você coloca a mão no código ou trabalhar nos processos de desenvolvimento de software ou ainda trabalhar diretamente com a gestão do todo. Ou seja, empreender ou trabalhar em times/empresas.
Já para quem faz Ciência da Computação pode fazer tudo que foi abordado acima, contudo visa mais a parte pesquisa / docência. 
Sou formado em Análise e Desenvolvimento de Sistemas. Confesso que entrei no curso extremamente perdido mas tinha dentro de mim a certeza que era o que queria fazer por toda vida. Sempre tive espirito empreendedor, participei de pesquisas cientificas durante o curso, fiz projetos visto e apresentado em congresso internacional, mas eu sempre tive dentro de mim o lado criador da coisa. Pensei que Ciência da Computação seria minha próxima graduação, más HOJE tenho certeza que se fosse evoluir em termos de conhecimento acadêmico / profissional o curso para meu perfil empreendedor, de ter ideias, inovar na criação de sistemas, trabalhar com times, eu faria o curso de Engenharia de Software.

**Comment by TheVoidAbyss** [comment]
Pessoal eu gosto muito de programação. Entre ciências da computação e Engenharia de software qual tem laços mais fortes com programação, ou melhor quem programa mais..

**Comment by wilsonneto-swe** [comment]
Não sei bem quanto a engenharia de software, eu fiz CC e programei muito pouco, mas aprendi conceitos e teorias computacionais, matemáticas e lógicos muito fortes que carrego comigo até hoje e acho que muitas das vezes é mais importante do que o código em si. 

Lembre-se que programação atualmente é bem auto nível e vejo como praticamente a ponta do iceberg, para graduação penso que você realmente precisa ver o que está por trás disso, o que vemos de desenvolvedores que não fazem ideia da complexidade ciclomática do código que escreve, assim como não fazem ideia do que acontece "abaixo do código"... é bom você saber como teu software lida com as camadas abaixo dele, com o processador, com a memória, como tudo ali funciona. 

Acredito que programar você aprende programando, vai se quando fizer alguns freelances e entrar no mercado como estagiário ou junior que vai realmente aprender e começar a lidar bem com código, mas agora o que acontece abaixo disso, isso sim penso ser a obrigação da faculdade te ensinar, pois o mercado não vai te ensinar, e por vezes vejo que isso diferencia muitos programadores no mercado. 

Apenas minha opinião,
abraços e sucesso!

---
Wilson Neto
Linkedin: [https://www.linkedin.com/in/wilsonnetobr/](https://www.linkedin.com/in/wilsonnetobr/)

**Comment by Peu255** [comment]
Isso sim é uma discussão útil. Obrigado pelas dicas e informações, pessoal.
